cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/loadbalancer/GcpBackendServiceResourceBuilder.java

```diff
@@ -126,11 +126,12 @@ private List<CloudResource> filterResourcesByType(List<CloudResource> resources,
                 .collect(Collectors.toList());
     }
 
-    private void makeBackendForTargetGroup(GcpContext context, CloudLoadBalancer loadBalancer, String projectId, String zone,
+    private void makeBackendForTargetGroup(GcpContext context, AuthenticatedContext auth, CloudLoadBalancer loadBalancer, String projectId, String zone,
             Set<Group> groups, List<Backend> backends) {
         for (Group group : groups) {
             Backend backend = new Backend();
-            String groupname = getResourceNameService().resourceName(ResourceType.GCP_INSTANCE_GROUP, context.getName(), group.getName());
+            String groupname = getResourceNameService()
```

Comment:
this id is probably the database index here. Lts not expose it to enduser but generate some kind of hash or anything.

Reply:
I am not a big fan of ID in data exposed to user but...
1) This strategy is already heavily used in resource generation for AWS (security groups, native instances, native cloud watch, disk tags) so it is good to keep consistent approach.
2) If we hash it - we will get a longer name which might be a problem since resource name length limitation. Short hash would increase a probability of collisions. Since the hash function is known it has zero cryptographic sense - we only use decimal numbers and create a dictionary for decimals for a keen "customer" is almost zero effort. Salting it would be overkill since salt then should be externalized.
3) Not sure about this one but this resource name is used for internal mapping between user-exposed group name and internal cloud-specific name. I am not sure if we even expose this one to a customer.